src/commands/force/org/beta/create.ts

```diff
@@ -269,9 +269,14 @@ export class Create extends SfdxCommand {
       this.logger.debug('Set Alias: %s result: %s', this.flags.setalias, result);
     }
     if (this.flags.setdefaultusername) {
-      const globalConfig: Config = this.configAggregator.getGlobalConfig();
-      globalConfig.set(Config.DEFAULT_USERNAME, username);
-      const result = await globalConfig.write();
+      let config: Config;
+      try {
+        config = await Config.create({ isGlobal: false });
+      } catch {
+        config = await Config.create({ isGlobal: true });
+      }
+      const result = config.set(Config.DEFAULT_USERNAME, username);
+      await config.write();
```

Comment:
When setting the default username during org create, we only want to set it locally, not globally.  It would be a breaking change if we set it globally.  I think I would just debug log if the config doesn't update successfully and maybe add warning output.  This change needs a test as well.

Reply:
I'm sorry!  I was wrong about setting the global default.  Since `org:create` doesn't require a project you can run the command from anywhere and if you do that, it'll fail to set the local default but set it globally.  This is the expected and desired behavior.